File: OUIKit/OUIMoments/src/main/java/io/openim/android/ouimoments/MainActivity.java
package io.openim.android.ouimoments;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.KeyEvent;

import io.openim.android.ouimoments.ui.CircleFragment;

public class MainActivity extends AppCompatActivity {
    CircleFragment fragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_moments_home);

        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        fragment = CircleFragment.newInstance(null);
        if (!fragment.isAdded()) {
            transaction.add(R.id.contentFl, fragment);
            transaction.show(fragment).commit();
        }

    }

    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (fragment.onKeyDown(keyCode, event)) return true;
        else return super.onKeyDown(keyCode, event);
    }

}

File: app/src/main/java/io/openim/demo/MainActivity.kt
package io.openim.demo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import cn.alvince.zanpakuto.databinding.ActivityBinding
import cn.alvince.zanpakuto.databinding.ActivityBindingHolder
import io.openim.demo.account.AccountFuncVisitor
import io.openim.demo.account.AccountManager
import io.openim.demo.databinding.MainActivityBinding
import io.openim.demo.home.fragment.HomeHostFragment

class MainActivity : AppCompatActivity(), ActivityBindingHolder<MainActivityBinding> by ActivityBinding(R.layout.main_activity) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        inflate {
            supportFragmentManager.beginTransaction()
                .add(R.id.fcv_container, HomeHostFragment())
                .commit()
        }
    }

    override fun onStart() {
        super.onStart()
        if (!AccountManager.logged) {
            AccountFuncVisitor.login(supportFragmentManager, R.id.fcv_container)
        }
    }
}
